### PR TITLE
Derive clone for hmac key

### LINKED ISSUE #806: Should hmac::SigningKey (also) implement Clone?
I see that `hmac::SigningKey`'s only field is of type `SigningContext` and it implements `Clone`. Is there a reason `hmac::SigningKey` does not implement `Clone`?

I can open a PR to add this if needed.

### PR DESCRIPTION
**Pull request by dvic**
This closes #806.

### CODE REVIEW

File: src/aead/aes.rs [hunk 1]
```diff
@@ -19,6 +19,7 @@ use super::{
 use crate::{bits::BitLength, cpu, endian::*, error, polyfill};
 use libc::size_t;
 
+#[derive(Clone)]
```
Comment (briansmith) [inline src/aead/aes.rs]: Please modify aead_test.rs to verify that the clone of a key produces the same results as the original key.

File: src/hmac.rs [hunk 2]
```diff
@@ -167,6 +167,7 @@ pub type Signature = Tag;
 pub struct Tag(digest::Digest);
 
 /// A key to use for HMAC signing.
+#[derive(Clone)]
```
Comment (briansmith) [inline src/hmac.rs]: Similarly, please modify hmac_test.rs to verify that the clone of a `Signing` (resp. `VerificationKey`) gives the same results as the original

**Comment by dvic** [comment]
I have updated `test_aead` and `hmac_test_case_inner` and added the commit message suffix. Let me know if anything else needs to be done 👍

File: src/aead/aes.rs (hunk 1 above)
Comment (dvic) [reply src/aead/aes.rs]: Done.

File: src/hmac.rs (hunk 2 above)
Comment (dvic) [reply src/hmac.rs]: Done.

**Comment by dvic** [comment]
🤦‍♂️ You are right, I could have guessed that (you have a comment about this in the code, see below). I'll remove the AEAD changes (HMAC was indeed what I initially requested) 👍. 

image

**Comment by dvic** [comment]
I have updated this PR for the latest changes related to merging into one `hmac::Key` and to remove the AEAD changes, as you suggested.

**Comment by briansmith** [comment]
Thank you!

**Pull request merged**